Sources/PackageDescription/PackageDependency.swift

```diff
@@ -221,44 +229,162 @@ extension Package.Dependency {
+        _ requirement: Package.Dependency.Requirement
     ) -> Package.Dependency {
-        return .init(name: name, url: path, requirement: .localPackageItem)
+        precondition(!requirement.isLocalPackage, "Use `.package(path:)` API to declare a local package dependency")
+        return .init(name: name, url: url, requirement: requirement)
+    }
+}
+
+// MARK: - registry
+
+extension Package.Dependency {
+    /// Adds a package dependency that uses the version requirement, starting with the given minimum version,
+    /// going up to the next major version.
+    ///
+    /// This is the recommended way to specify a remote package dependency.
+    /// It allows you to specify the minimum version you require, allows updates that include bug fixes
+    /// and backward-compatible feature updates, but requires you to explicitly update to a new major version of the dependency.
+    /// This approach provides the maximum flexibility on which version to use,
+    /// while making sure you don't update to a version with breaking changes,
+    /// and helps to prevent conflicts in your dependency graph.
+    ///
+    /// The following example allows the Swift Package Manager to select a version
+    /// like a  `1.2.3`, `1.2.4`, or `1.3.0`, but not `2.0.0`.
+    ///
+    ///    .package(identity: "scope/name", from: "1.2.3"),
+    ///
+    /// - Parameters:
+    ///     - identity: The identity of the package.
+    ///     - version: The minimum version requirement.
+    @available(_PackageDescription, introduced: 999)
+    public static func package(
+        identity: String,
+        from version: Version
+    ) -> Package.Dependency {
+        return .package(identity: identity, .upToNextMajor(from: version))
+    }
+
+    /// Adds a remote package dependency given a branch requirement.
+    ///
+    ///    .package(identity: "scope/name", branch: "main"),
+    ///
+    /// - Parameters:
+    ///     - identity: The identity of the package..
+    ///     - branch: A dependency requirement. See static methods on `Package.Dependency.Requirement` for available options.
+    @available(_PackageDescription, introduced: 999)
+    public static func package(
+        identity: String,
+        branch: String
+    ) -> Package.Dependency {
+        return .package(identity: identity,  .branch(branch))
+    }
+
+    /// Adds a remote package dependency given a revision requirement.
+    ///
+    ///    .package(identity: "scope/name", revision: "aa681bd6c61e22df0fd808044a886fc4a7ed3a65"),
+    ///
+    /// - Parameters:
+    ///     - identity: The identity of the package.
+    ///     - revision: A dependency requirement. See static methods on `Package.Dependency.Requirement` for available options.
+    @available(_PackageDescription, introduced: 999)
+    public static func package(
+        identity: String,
+        revision: String
+    ) -> Package.Dependency {
+        return .package(identity: identity, .revision(revision))
+    }
```

Comment:
Dependencies on on a package in a registry can only be pinned to a range of versions (that is, no branches or revisions).

```suggestion
```